src/HVX/Primitives.hs

```diff
@@ -53,13 +55,28 @@ infixl 6 +~
 hmul :: (ApplyVex 'Affine 'Nonmon v1 m1 ~ v2, ValidVex v2)
   => Expr 'Affine 'Const -> Expr v1 m1 -> Expr v2 (ApplyMon 'Nonmon m1)
 hmul (EConst a) e = apply (Mul a) e
-hmul _ _ = error "the left argument of a multiply must be a constant"
+hmul _ _ = error "The left argument of a multiply must be a constant"
 
 infixl 7 *~
 (*~) :: (ApplyVex 'Affine 'Nonmon v1 m1 ~ v2, ValidVex v2)
   => Expr 'Affine 'Const -> Expr v1 m1 -> Expr v2 (ApplyMon 'Nonmon m1)
 (*~) (EConst a) e = apply (Mul a) e
-(*~) _ _ = error "the left argument of a multiply must be a constant"
+(*~) _ _ = error "The left argument of a multiply must be a constant"
+
+hmulpos :: (ApplyVex 'Affine 'Nondec v1 m1 ~ v2, ValidVex v2)
+  => Expr 'Affine 'Const -> Expr v1 m1 -> Expr v2 (ApplyMon 'Nondec m1)
+hmulpos (EConst a) e
+  | 0 <= minElement a = apply (MulPos a) e
+  | otherwise = error "The left argument of a left-positive multiply must have non-negative entries"
```

Comment:
Ok so the issue i'm trying to solve here is that when a non-positive entry is entered, we see an error like this:
```
> solve 1
([("x1",(1><1)
 [ 10.0 ]),("x2",(1><1)
 [ 200.0 ]),("x3",(1><1)
 [ 400.0 ]),("lineVar",(1><1)
 [ 10.0 ])],*** Exception: The left argument of a left-positive multiply must have non-negative entries
CallStack (from HasCallStack):
  error, called at src/HVX/Primitives.hs:78:17 in hvx-0.3.1.0-HnMwpnJtEwx2GYI4IYHzYJ:HVX.Primitives
```
so we need to identify the function throwing the error, otherwise the user won't know why the error is happening and will be unable to debug it. We also identify the reason the function is throwing the error, so that the user doesn't have to look at the `hmulpos` source code to figure that out.

How would you do both without being redundant but maintaining a clear description of what went wrong and where?